Pass toolbar button ID in `wxContextHelpEvent`

StevenPHirshman opened the pull request:
Modified cshelp.cpp file so that context help click on toolbar button returns button ID, NOT toolbar ID.

StevenPHirshman committed:
Update cshelp.cpp

StevenPHirshman committed:
Merge branch 'master' of https://github.com/StevenPHirshman/wxWidgets

StevenPHirshman commented:
Strange – I was able to build the library with no errors. Any suggestion WHERE the virtual function could go?

***@***.***

From: VZ ***@***.***>
Sent: Thursday, September 18, 2025 7:36 AM
To: wxWidgets/wxWidgets ***@***.***>
Cc: StevenPHirshman ***@***.***>; Author ***@***.***>
Subject: Re: [wxWidgets/wxWidgets] fix for cshelp.cpp (PR #25809)

[Image removed by sender.]vadz left a comment (wxWidgets/wxWidgets#25809)<https://github.com/wxWidgets/wxWidgets/pull/25809#issuecomment-3306976225>

Thanks, but I'd really prefer to have a virtual function instead of using dynamic cast like this.

And, of course, as you can see from the CI failures, the current version doesn't compile due to missing wx/toolbar.h include (but it won't be necessary if/when you change the code to use the virtual function).

—
Reply to this email directly, view it on GitHub<https://github.com/wxWidgets/wxWidgets/pull/25809#issuecomment-3306976225>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/BNH2DLHVVLMSTBFXF7WTBKD3TKKJ7AVCNFSM6AAAAACG2AV5SSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTGMBWHE3TMMRSGU>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

StevenPHirshman committed:
added virtual SetHelpEventPos

added virtual function SetHelpEventPos to handle toolbar help clicks

vadz reviewed (changes requested):
Thanks, I don't really understand what is the advantage of doing it like this rather than using `GetHelpIdAtPoint()` as I thought to do it, but why not.

The real problem for me is that:

1. This function name is misleading, it sets the ID and (not just the) position.
2. IMO it shouldn't set the position at all, it doesn't make much sense to have it in client coords for the toolbars and screen coords for everything else.

So could you please rename the function to something like `AdjustHelpEvent()` and remove `SetPosition()` call from it unless there is some good reason to do it that I'm missing?

StevenPHirshman commented:
I’ve changed the method name to SetHelpIdAtPoint. It change the event id to the button clicked, and not the toolbar id. No need to change the position.

***@***.***

From: VZ ***@***.***>
Sent: Thursday, September 18, 2025 11:46 AM
To: wxWidgets/wxWidgets ***@***.***>
Cc: StevenPHirshman ***@***.***>; Author ***@***.***>
Subject: Re: [wxWidgets/wxWidgets] fix for cshelp.cpp (PR #25809)


@vadz requested changes on this pull request.

Thanks, I don't really understand what is the advantage of doing it like this rather than using GetHelpIdAtPoint() as I thought to do it, but why not.

The real problem for me is that:

  1.  This function name is misleading, it sets the ID and (not just the) position.
  2.  IMO it shouldn't set the position at all, it doesn't make much sense to have it in client coords for the toolbars and screen coords for everything else.

So could you please rename the function to something like AdjustHelpEvent() and remove SetPosition() call from it unless there is some good reason to do it that I'm missing?

________________________________

In include/wx/window.h<https://github.com/wxWidgets/wxWidgets/pull/25809#discussion_r2359968878>:

> @@ -1448,6 +1448,9 @@ class WXDLLIMPEXP_CORE wxWindowBase : public wxEvtHandler

         return GetHelpTextAtPoint(wxDefaultPosition, wxHelpEvent::Origin_Unknown);

     }



+       //sph091825

Please don't do this, we use version control for this. Just imagine what this file would look like if each of hundreds of contributors to it left such lines in it.
⬇️ Suggested change

-       //sph091825

+       //sph091825

________________________________

In include/wx/window.h<https://github.com/wxWidgets/wxWidgets/pull/25809#discussion_r2359970485>:

> @@ -1448,6 +1448,9 @@ class WXDLLIMPEXP_CORE wxWindowBase : public wxEvtHandler

         return GetHelpTextAtPoint(wxDefaultPosition, wxHelpEvent::Origin_Unknown);

     }



+       //sph091825

+       virtual void SetHelpEventPos(wxHelpEvent& WXUNUSED(event)) { return; }

This is trivial, but why have this useless return?
⬇️ Suggested change

-       virtual void SetHelpEventPos(wxHelpEvent& WXUNUSED(event)) { return; }

+       virtual void SetHelpEventPos(wxHelpEvent& WXUNUSED(event)) { }

—
Reply to this email directly, view it on GitHub<https://github.com/wxWidgets/wxWidgets/pull/25809#pullrequestreview-3240576969>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/BNH2DLFMYD4XVZYZTMSINHT3TLHTVAVCNFSM6AAAAACG2AV5SSVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZTENBQGU3TMOJWHE>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

StevenPHirshman committed:
Replace SetHelpEventPos with SetHelpIdAtPoint

vadz commented:
Please reset the draft status when the builds pass.

BTW, I forgot to mention this, but it's probably worth mentioning this behaviour in `wxHelpEvent` documentation in `interface/wx/event.h`. TIA!

StevenPHirshman commented:
What do I need to do about some of the build error messages, like this one:

Mac builds / wxiOS
Failed in 13 minutes and 44 seconds

***@***.***

From: VZ ***@***.***>
Sent: Thursday, September 18, 2025 12:45 PM
To: wxWidgets/wxWidgets ***@***.***>
Cc: StevenPHirshman ***@***.***>; Author ***@***.***>
Subject: Re: [wxWidgets/wxWidgets] Pass toolbar button ID in `wxContextHelpEvent` (PR #25809)

[Image removed by sender.]vadz left a comment (wxWidgets/wxWidgets#25809)<https://github.com/wxWidgets/wxWidgets/pull/25809#issuecomment-3308505948>

Please reset the draft status when the builds pass.

BTW, I forgot to mention this, but it's probably worth mentioning this behaviour in wxHelpEvent documentation in interface/wx/event.h. TIA!

—
Reply to this email directly, view it on GitHub<https://github.com/wxWidgets/wxWidgets/pull/25809#issuecomment-3308505948>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/BNH2DLA76BSH4YFLHNGBMYL3TLOSJAVCNFSM6AAAAACG2AV5SSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTGMBYGUYDKOJUHA>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

StevenPHirshman committed:
added comments to window.h and event.h

StevenPHirshman committed:
updated information text in event.h

StevenPHirshman commented:
Updated the wxHelpEvent document and removed any comments in the code (which I had put there so I could easily find the modified sections).

Let me know anything else is required.

***@***.***

From: VZ ***@***.***>
Sent: Thursday, September 18, 2025 12:45 PM
To: wxWidgets/wxWidgets ***@***.***>
Cc: StevenPHirshman ***@***.***>; Author ***@***.***>
Subject: Re: [wxWidgets/wxWidgets] Pass toolbar button ID in `wxContextHelpEvent` (PR #25809)

[Image removed by sender.]vadz left a comment (wxWidgets/wxWidgets#25809)<https://github.com/wxWidgets/wxWidgets/pull/25809#issuecomment-3308505948>

Please reset the draft status when the builds pass.

BTW, I forgot to mention this, but it's probably worth mentioning this behaviour in wxHelpEvent documentation in interface/wx/event.h. TIA!

—
Reply to this email directly, view it on GitHub<https://github.com/wxWidgets/wxWidgets/pull/25809#issuecomment-3308505948>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/BNH2DLA76BSH4YFLHNGBMYL3TLOSJAVCNFSM6AAAAACG2AV5SSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTGMBYGUYDKOJUHA>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

StevenPHirshman committed:
updated line endings

StevenPHirshman committed:
Change CR/LF to LF and add terminating LF

StevenPHirshman committed:
Delete include/wx/window.h

StevenPHirshman committed:
Delete src/common/cshelp.cpp

StevenPHirshman committed:
Delete src/common/tbarbase.cpp

StevenPHirshman committed:
Delete src/common/wincmn.cpp

StevenPHirshman committed:
Delete include/wx/tbarbase.h

StevenPHirshman committed:
Merge branch 'StevenPHirshman'

vadz commented:
Just for the recorded, you (or your editor) have completely reformatted all the files you have touched. You need to configure it to prevent this from happening.

Please make the changes in your fork of the repository, confirm that they look good to you by comparing them with this one and then, and only then, push them here.

Thanks in advance!

StevenPHirshman committed:
CR/LF Mods for SetHelpIdAtPoint

StevenPHirshman committed:
LR?CR fix 2

StevenPHirshman commented:
> Just for the recorded, you (or your editor) have completely reformatted all the files you have touched. You need to configure it to prevent this from happening.
> 
> Please make the changes in your fork of the repository, confirm that they look good to you by comparing them with this one and then, and only then, push them here.
> 
> Thanks in advance!

I updated the files. Now there are no EOL/White Space issues, but ALL compilation tests are failing now. I don't understand why that is happening.
